### DISCUSSION TITLE
Can OPA compare two JSON files and capture the differences between two JSON file into another

### DISCUSSION THREAD

**Discussion by sbalakrishn** [open-policy-agent/community#284] [OPA and Rego]
Hi 

Rego code..that can help us to compare two JSON files and find the differences between the two into another JSON file

We are trying to compare two JSON  files and identify the difference between the two JSON file . Can OPA help us in capturing the difference between the two JSON files into another file.

Regards
Sreejith Balakrishnan

**Comment by anderseknert** [comment]
Hi @sbalakrishn! I don't think there's any built-in function that allows you to do this without some manual work. See https://github.com/open-policy-agent/opa/issues/1143 for an open issue on this.

I guess I would do something like traverse each object using the `walk` built-in function, and compare it to the corresponding node of the other object.. but could be tricky to report it back in a nice format. If you have an example of two objects to compare, we could give it a shot :)
